web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -0,0 +1,293 @@
+<template>
+  <div class="ui floating filter dropdown custom">
+    <button class="branch-dropdown-button gt-ellipsis ui basic small compact button gt-df" @click="menuVisible = !menuVisible" @keyup.enter="menuVisible = !menuVisible">
+      <span class="text gt-df gt-ac gt-mr-2">
+        <template v-if="release">{{ textReleaseCompare }}</template>
+        <template v-else>
+          <svg-icon v-if="isViewTag" name="octicon-tag" />
+          <svg-icon v-else name="octicon-git-branch"/>
+          <strong ref="dropdownRefName" class="gt-ml-3">{{ refNameText }}</strong>
+        </template>
+      </span>
+      <svg-icon name="octicon-triangle-down" :size="14" class-name="dropdown icon"/>
+    </button>
+    <div class="menu transition" :class="{visible: menuVisible}" v-if="menuVisible" v-cloak>
+      <div class="ui icon search input">
+        <i class="icon gt-df gt-ac gt-jc gt-m-0"><svg-icon name="octicon-filter" :size="16"/></i>
+        <input name="search" ref="searchField" autocomplete="off" v-model="searchTerm" @keydown="keydown($event)" :placeholder="searchFieldPlaceholder">
+      </div>
+      <template v-if="showBranchesInDropdown">
```

Comment:
Again, why so many nested `template`s?

Reply:
To keep the same code structure like the tmpl